src/Analyser/NodeScopeResolver.php

```diff
@@ -1465,6 +1465,23 @@ function (MutatingScope $scope) use ($expr, $nodeCallback, $context): Expression
 			) {
 				$scope = $scope->assignVariable('http_response_header', new ArrayType(new IntegerType(), new StringType()));
 			}
+
+			if (
+				isset($functionReflection)
+				&& $functionReflection->getName() === 'extract'
+			) {
+				if (
+					$scope->getType($expr->args[0]->value) instanceof ConstantArrayType
+				) {
+					$extract = $scope->getType($expr->args[0]->value);
```

Comment:
There are 2 more possible Parameters here. I'd like to Support only the default parameters,
but had problems here.

I get `ConstantType` as second param, but one could also drop the corresponding `int` value in.
How do I check for that?

```php
extract(array &$array [,int $flags=EXTR_OVERWRITE [, string $prefix=NULL]] ) : int
```

Reply:
$flags = $scope->getType($expr->args[1]->value);
but you need to check if there is some argument, or example via `count($expr->args) >= 2`

Reply:
should be solved now.

I did a quick research, how often the `flags`-Attibutes are beeing used.


supported | const | usages | url |
| ---------- | ------ | --------| --- |
| [x]  | `EXTR_OVERWRITE` | ∞ |  its the default |
| [ ] | `EXTR_SKIP` | 675,812 | https://github.com/search?l=PHP&q=EXTR_SKIP&type=Code
| [ ] | `EXTR_PREFIX_ALL` | 34,886 | https://github.com/search?l=PHP&q=EXTR_PREFIX_ALL&type=Code
| [ ] | `EXTR_PREFIX_INVALID` | 3,321 | https://github.com/search?l=PHP&q=EXTR_PREFIX_INVALID&type=Code
| [ ] | `EXTR_IF_EXISTS` | 21,810 | https://github.com/search?l=PHP&q=EXTR_IF_EXISTS&type=Code
| [ ] | `EXTR_PREFIX_IF_EXISTS` | mostly tests| ...
| [ ] | `EXTR_REFS` | mostly tests | https://github.com/search?l=PHP&q=EXTR_REFS&type=Code
---

Since it's possible to combine the constants, many possible use cases exist.
For my case, only `EXP_OVERWRITE` ist relevant, so I implemented only this case.

If you want, I could provide support for `EXPR_SKIP` and `EXTR_PREFIX_ALL`, too, since they seem to be widely used (but could also go into an other PR).

I just don't want to support combinations like `EXTR_OVERWRITE | EXTR_REFS`.

WDYT?